src/acvp_util.c

```diff
@@ -462,6 +464,76 @@ ACVP_RSA_CAP_MODE_LIST* acvp_locate_rsa_mode_entry(ACVP_CAPS_LIST *cap,
     return NULL;
 }
 
+ACVP_RSA_CAP_MODE_LIST* acvp_locate_rsa_sig_type_entry(ACVP_CAPS_LIST *cap,
```

Comment:
Just wondering if there should be a distinction between the return cases? Seems like the 'error' case and the 'not found' cases could be treated the same way. So sig_type_str error or others are treated as 'not found' which could cause an attempt to allocate a linked list entry.

Maybe we could define and create a dummy 'empty struct' when the object is not found and use 'null' for the error case?

Reply:
Good catch it would make a new mode list with the bogus mode type. I am not sure which way we want to do things but I was planning on having that function say it either found what you were looking for or it didn't. I think that we should pull the mode string check outside of the search function and error there and keep that function simply as a search.